### PR TITLE
fix: uniq by hash, instead of transaction

### PR DESCRIPTION
**Pull request by zachdaniel**
#883 

### Bug Fixes
* when deduplicating transactions, don't compare the entire object, because preloaded associations could have the list objects in a different order. Instead, deduplicate them using their hash. 

  - [x] I added an entry to `CHANGELOG.md` with this PR
  - [x] If I added new functionality, I added tests covering it.
  - [ ] If I fixed a bug, I added a regression test to prevent the bug from silently reappearing again.
  - [x] I checked whether I should update the docs and did so if necessary

I tried really hard to reproduce this in a test, but the problem is that it is an inconsistent behavior. Postgres chooses the order of the returned rows. I could sort them to ensure that they always come back in the same order, but that is not actually a requirement of the UI, so we would be slowing it down just to be able to test it. I don't mind adding the sort for the sake of the test, just let me know. 

I'm pretty positive that this is the problem but I haven't been able to find a local occurrence of this myself so we should test it against the sokol instance that had the original issue reported.

### CODE REVIEW

File: apps/explorer/lib/explorer/chain/token_transfer.ex [hunk 1]
```diff
@@ -204,59 +204,26 @@ defmodule Explorer.Chain.TokenTransfer do
...
-
-          SELECT transaction_hash
-          FROM token_transfers
-          WHERE to_address_hash = $1
-          ) as token_transfers_transaction_hashes
-          LIMIT $2
-        """,
-        [address_bytes, page_size]
+  defp transaction_hashes_from_token_transfers_sql(address_bytes, %PagingOptions{page_size: page_size} = paging_options) do
+    to_query =
```
Comment (zachdaniel) [inline apps/explorer/lib/explorer/chain/token_transfer.ex]: This doesn't need to be changed, but while debugging this issue I cleaned up this query. Might as well use the new more legible version. I will remove this change on request.

File: apps/explorer/lib/explorer/chain/token_transfer.ex [hunk 2]
```diff
@@ -204,59 +204,16 @@ defmodule Explorer.Chain.TokenTransfer do
...
-          FROM token_transfers
-          WHERE to_address_hash = $1
-          AND block_number < $2
-          ) as token_transfers_transaction_hashes
-          LIMIT $3
-        """,
-        [address_bytes, block_number, page_size]
-      )
+  defp transaction_hashes_from_token_transfers_sql(address_bytes, %PagingOptions{page_size: page_size} = paging_options) do
+    query =
```
Comment (zachdaniel) [inline apps/explorer/lib/explorer/chain/token_transfer.ex]: This doesn't need to be changed, but while debugging this issue I cleaned up this query. Might as well use the new more legible version. I will remove this change on request.

**Comment by vbaranov** [comment]
>I'm pretty positive that this is the problem but I haven't been able to find a local occurrence of this myself so we should test it against the sokol instance that had the original issue reported.

I can reproduce this issue in stg. I will test this fix with the new wave of updates to stg.

**Comment by vbaranov** [comment]
>I can reproduce this issue in stg. I will test this fix with the new wave of updates to stg.

I hurried with this statement. I didn't find an example on stg. I will see this fix live on Sokol.

**Pull request merged**